data/pigui/modules/station-view/02-bulletinBoard.lua

```diff
@@ -56,6 +56,10 @@ local function refresh()
 
 	for ref,ad in pairs(ads or {}) do
 		if searchText == ""
+			or searchText ~= "" and string.find(
```

Comment:
Isn't `searchText ~= "" and` in this and the next paragraph redundant after the test on line 58?

Reply:
I mean you could simplify it as such:

```lua
        for ref,ad in pairs(ads or {}) do
                if searchText == ""
-                       or searchText ~= "" and string.find(
+                       or string.find(
                                string.lower(ad.title),
                                string.lower(searchText),
                                1, true)
-                       or searchText ~= "" and string.find(
+                       or string.find(
                                string.lower(ad.description),
                                string.lower(searchText),
                                1, true)

```

Reply:
@zonkmachine is correct, `searchText ~= ""` is implied by the first condition of the if statement evaluating to false. The only way this breaks is if `nil` is passed as `searchText`, which can be handled by changing the first condition to `if not searchText or searchText == ""`.